dev.exs

```diff
@@ -28,6 +28,80 @@ Application.put_env(:phoenix_live_dashboard, DemoWeb.Endpoint,
   ]
 )
 
+defmodule DemoWeb.History do
+  use GenServer
+  alias Phoenix.LiveDashboard.TelemetryListener
+
+  @history_buffer_size 50
+
+  def data(metric) do
+    GenServer.call(__MODULE__, {:data, metric})
+  end
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @impl true
+  def init(metrics) do
+    Process.flag(:trap_exit, true)
+    GenServer.cast(__MODULE__, {:metrics, metrics})
+    {:ok, %{}}
+  end
+
+  @impl true
+  def terminate(_, events) do
+    for event <- events do
+      :telemetry.detach({__MODULE__, event, self()})
+    end
+
+    :ok
+  end
+
+  defp attach_handler(%{name: name_list} = metric, id) do
+    :telemetry.attach(
+      "#{inspect(name_list)}-history-#{id}",
```

Comment:
The handler name we are attaching is not the same we are detaching in the function above.

Reply:
For clarify, I would propose for the whole metric to be part of the name. So here we should do:

    {__MODULE__, metric, self()}

On terminate it should be:

```
    for {metric, _} <- metrics do
      :telemetry.detach({__MODULE__, metric, self()})
    end
```